### DISCUSSION TITLE
Overwrite `toSelectOption` with manually defining the label and/or value fields

### DISCUSSION THREAD

**Discussion by jhhazelaar** [spatie/laravel-options#7] [Ideas]
Hi, thank you for creating this packages!

We are using it with different models and all implemented the `Selectable` with `toSelectOption`. This works fine with all the models, but we have a country model with name, iso2 and phone_code als fields. For a country select with the `toSelectOption` option like below works great:

```php
public function toSelectOption(): SelectOption
{
    return new SelectOption(
        $this->id,
        $this->name
    );
}
```

When creating a country code select with the phone code becomes a problem, because we wan't to overrule the `toSelectOption` in a template like this:

```php
\Spatie\LaravelOptions\Options::forModels(\App\Models\Country::class, value: 'iso2', label: 'phone_code')->toArray()
```

Is it an idea to pass the value and label property to `toSelectOption` as arguments so you can use them in de `toSelectOption` value?

Hopefully my explanation is any good :-)

**Comment by rubenvanassche** [comment]
I'm not completely following along, how would toSelectOption work in this case? And how would you construct an options object using it?
